spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java

```diff
@@ -133,32 +137,15 @@ else if (this.flywayDataSource != null) {
 			else {
 				flyway.setDataSource(this.dataSource);
 			}
+			if (!CollectionUtils.isEmpty(this.flywayCallbacks)) {
+				for (FlywayCallback flywayCallback : this.flywayCallbacks) {
+					flyway.setCallbacks(flywayCallback);
+				}
+			}
 			flyway.setLocations(this.properties.getLocations().toArray(new String[0]));
 			return flyway;
 		}
 
-		@Bean
-		@ConditionalOnMissingBean
-		public FlywayMigrationInitializer flywayInitializer(Flyway flyway) {
-			return new FlywayMigrationInitializer(flyway, this.migrationStrategy);
-		}
-
-		/**
-		 * Additional configuration to ensure that {@link EntityManagerFactory} beans
-		 * depend-on the {@code flywayInitializer} bean.
-		 */
-		@Configuration
-		@ConditionalOnClass(LocalContainerEntityManagerFactoryBean.class)
-		@ConditionalOnBean(AbstractEntityManagerFactoryBean.class)
-		protected static class FlywayInitializerJpaDependencyConfiguration
```

Comment:
We'll still need something that's the equivalent of this to ensure that Flyway migrations have been run for Hibernate is initialized.